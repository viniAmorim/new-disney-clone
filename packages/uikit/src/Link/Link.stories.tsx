import React from 'react';

import { Story, Meta } from '@storybook/react';

import Link, { LinkProps } from '~/Link';

export default {
  title: 'Form/Link',
  component: Link,
} as Meta;

const Template: Story<LinkProps> = (args) => <Link {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Esqueci a senha',
};
