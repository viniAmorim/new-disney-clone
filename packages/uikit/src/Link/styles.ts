import styled from 'styled-components';

export const Link = styled.a`
  color: ${({ theme }) => theme.colors['blue-200']};
  cursor: pointer;
  text-decoration: none;
  ${({ theme }) => theme.typography['subheader-light']};

  &:hover,
  &:active {
    color: ${({ theme }) => theme.colors['blue-400']};
  }
`;
