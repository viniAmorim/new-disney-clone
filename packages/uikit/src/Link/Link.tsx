import React, { BaseHTMLAttributes, forwardRef, ReactNode } from 'react';

import * as Styled from './styles';

export interface LinkProps extends BaseHTMLAttributes<HTMLAnchorElement> {
  /**
   * Link content.
   */
  children: ReactNode;
}

const Link = forwardRef<any, LinkProps>(
  ({ children, ...rest }: LinkProps, ref) => {
    return (
      <Styled.Link ref={ref} {...rest}>
        {children}
      </Styled.Link>
    );
  },
);

export default Link;
