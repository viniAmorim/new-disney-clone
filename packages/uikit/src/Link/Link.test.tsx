import React from 'react';

import { render } from '~/test/utils';

import Link from './Link';

describe('Link', () => {
  it('should render correctly', () => {
    const { getByRole } = render(<Link href="https://correta.app">Link</Link>);

    const linkElement = getByRole('link');

    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveTextContent(/^Link$/);

    expect(linkElement).toMatchSnapshot();
  });
});
