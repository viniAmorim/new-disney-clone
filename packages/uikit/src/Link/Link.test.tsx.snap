// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<a
  class="sc-bdfBQB fsnmib"
  href="https://correta.app"
>
  Link
</a>
`;
